graph.py

```diff
@@ -0,0 +1,71 @@
+from itertools import product
+import random
+from collections import deque
+
+class Vertex():
+    def __init__(self,name:str,value=None,children:list=None):
+        if name is not None:
+            self.name = name
+        if value is not None:
+            self.value = value
+        else:
+            self.value = None
+
+        if children is not None:
+            self.children = children
+        else:
+            self.children = []
```

Comment:
Why `list` instead of `set`? If we want to do queries against the children `set` provides `O(1)` random access in the average case and `O(n)` in the worst, while `list` provides `O(n)` in the average case.